### PR TITLE
feat: add invalidate cache step

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Add invalidate cache step.
```
///                       ┌───────────────────┐
///                       │RegionFailoverStart│
///                       └─────────┬─────────┘
///                                 │
///                                 │ Selects a candidate(Datanode)
///                  ┌─────────┐    │ to place the failed region
///                  │         │    │
///  If replied with │     ┌───▼────▼───────┐
///  "Close region   │     │DeactivateRegion│
///  failed"         │     └───┬────┬───────┘
///                  │         │    │
///                  └─────────┘    │ Sends "Close Region" request
///                                 │ to the failed Datanode, and
///                  ┌─────────┐    │ wait for 2 seconds
///                  │         │    │
///                  │      ┌──▼────▼──────┐
/// Wait candidate   │      │ActivateRegion◄───────────────────────┐
/// response timeout │      └──┬────┬──────┘                       │
///                  │         │    │                              │
///                  └─────────┘    │ Sends "Open Region" request  │
///                                 │ to the candidate Datanode,   │
///                                 │ and wait for 30 seconds      │
///                                 │                              │
///                                 │ Check Datanode returns       │
///                                 │                              │
///                         success ├──────────────────────────────┘
///                                 │                       failed
///                       ┌─────────▼──────────┐
///                       │UpdateRegionMetadata│
///                       └─────────┬──────────┘
///                                 │
///                                 │ Updates the Region
///                                 │ placement metadata
///                                 │
///                         ┌───────▼───────┐
///                         │InvalidateCache│
///                         └───────┬───────┘
///                                 │
///                                 │ Broadcast Invalidate Table
///                                 │ Cache
///                                 │
///                                 │
///                        ┌────────▼────────┐
///                        │RegionFailoverEnd│
///                        └─────────────────┘
```

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#1126

### CODE REVIEW

File: src/meta-srv/src/procedure/region_failover/invalidate_cache.rs [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+                };
+
+                if result {
+                    Ok(())
+                } else {
+                    let reason = format!(
+                        "table cache {table_ident:?} is not invalidated by Frontend {}, error: {error:?}",
+                        ch,
+                    );
+                    error::RetryLaterSnafu { reason }.fail()
```
Comment (MichaelScofield) [inline src/meta-srv/src/procedure/region_failover/invalidate_cache.rs:89]: I don't think we need to retry here. Table routes in Frontend nodes are just cache, they can be updated periodically or upon some failures. Here adding the step of invalidate table cache in Frontend in region failover procedure is merely an optimization to client reads/writes, not a necessity. 

Even more, I don't think Meta should wait for Frontend's reply, just "fire and forget".

File: src/meta-srv/src/procedure/region_failover.rs [hunk 2]
```diff
@@ -227,6 +228,17 @@ trait State: Sync + Send + Debug {
...
 ///                        ┌────────▼────────┐
+///                        │ InvalidateCache ◄─────────────────────┐
+///                        └────────┬────────┘                     │
+///                                 │                              │
+///                                 │ Sends "Invalidate Table      │
+///                                 │ Cache" to Frontend nodes     │
+///                                 │ and waits for 2 seconds      │
+///                                 │                              │
+///                         success ├──────────────────────────────┘
+///                                 │                       failed
```
Comment (MichaelScofield) [inline src/meta-srv/src/procedure/region_failover.rs:239]: Now no retry and no wait, doc should be updated.

File: src/meta-srv/src/procedure/region_failover/invalidate_cache.rs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            .next(&context, &failed_region)
+            .await
+            .unwrap();
+        assert_eq!(format!("{next_state:?}"), "RegionFailoverEnd");
+
+        let frontend_id = |idx: usize| -> u64 { (idx + 4) as u64 };
+
+        for idx in 0..4 {
+            // frontend id starts from 4
+            let rx = heartbeat_receivers.get_mut(&frontend_id(idx)).unwrap();
```
Comment (MichaelScofield) [inline src/meta-srv/src/procedure/region_failover/invalidate_cache.rs:117]:
```suggestion
        for i in 4..=7 {
            // frontend id starts from 4
            let rx = heartbeat_receivers.get_mut(&i).unwrap();
```

**Review by MichaelScofield — COMMENTED** [review]
Please resolve the conflicts.

**Comment by WenyXu** [comment]
PTAL @MichaelScofield

**Review by MichaelScofield — APPROVED** [review]

**Comment by WenyXu** [comment]
PTAL @fengjiachun @killme2008

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**